### PR TITLE
DRM Support framework

### PR DESCRIPTION
**Pull request by tinusneethling**
This is support for generic DRM within the framework. Support for Android ClearKey as first test case provided. Below the methods to generate an encrypted media container file that is supported by ExpoPlayer2.

The audio channel from mp3 must be added to an MP4 file.
ffmpeg -i dataclear.mp3  -c copy dataclear.mp4

The new mp4 is encrypted using MP4Box
MP4Box -crypt drm_file_music.xml  dataclear.mp4  -out data.mp4

where drm_file_music.xml is:

 `
<GPACDRM type="CENC AES-CTR">
  <DRMInfo type="pssh" version="1">
    <BS ID128="1077efecc0b24d02ace33c1e52e2fb4b"/>
    <BS bits="32" value="1"/>
    <BS ID128="cd7eb9ff88f34caeb06185b00024e4c2"/>
  </DRMInfo>
  <CrypTrack IV_size="8" first_IV="0xbb5738fe08f11341" isEncrypted="1" saiSavedBox="senc" trackID="1">
    <key KID="f3c5e0361e6654b28f8049c778b23946" value="69eaa802a6763af979e8d1940fb88392"/>
  </CrypTrack>
 </GPACDRM>
 `

ExoPlayer2 only support Fragmented MP4 files. Therefore the mp4 file needs to be fragmented.

MP4Box -frag 240000 data.mp4 -out final.mp4

Audio.file can be called with drmConfiguration 

DrmConfiguration.clearKey(keys: {
            "f3c5e0361e6654b28f8049c778b23946":
            "69eaa802a6763af979e8d1940fb88392"
          })

or alternatively 

DrmConfiguration.clearKey(clearKey: '{"keys":[{"kty":"oct","kid":"88XgNh5mVLKPgEnHeLI5Rg","k":"aeqoAqZ2Ovl56NGUD7iDkg"}],"type":"temporary"}')

### CODE REVIEW

**Comment by kalismeras61** [comment]
@tinusneethling  did you test without DRM ? it will not blocking anything right ? If yes i will merge it.

**Comment by tinusneethling** [comment]
Hi,

Yes I have tested it. It will also only be used if parameters are sent.

And thank your for great piece of code.

Regards,
Tinus

**Comment by kalismeras61** [comment]
Thanks, last thing can you fix conflicts ?
